tests/integration_tests.py

```diff
@@ -647,6 +647,350 @@ def test_will_not_output_empty_state(self):
+        target_stitch.OUR_SESSION = FakeSession(mock_in_order_all_200)
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 1, "name": "Mike"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 2, "name": "Paul"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 3, "name": "Harrsion"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 4, "name": "The byte limit should be across streams, so lets make lots of data on both streams"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 5, "name": "to force the target to exceed its byte limit"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 6, "name": "A"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 7, "name": "B"}}))
+        # Should flush here
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "zebra_stream", "record": {"id": 8, "name": "C"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 9, "name": "D"}}))
+        self.queue.append(json.dumps({"type": "RECORD", "stream": "chicken_stream", "record": {"id": 10, "name": "E"}}))
+        # Should flush here
+
+        self.target_stitch.consume(self.queue)
+        finish_requests()
+
+        expected_messages = [
+            [{'action': 'upsert', 'data': {'id': 1, 'name': 'Mike'}},
+             {'action': 'upsert', 'data': {'id': 3, 'name': 'Harrsion'}},
+             {'action': 'upsert',
+              'data': {'id': 5,
+                       'name': 'to force the target to exceed its byte limit'}}],
+            [{'action': 'upsert', 'data': {'id': 2, 'name': 'Paul'}},
+             {'action': 'upsert', 'data': {'id': 4, 'name': 'The byte limit should be across streams, so lets make lots of data on both streams'}}],
+            [{'action': 'upsert', 'data': {'id': 6, 'name': 'A'}},
+             {'action': 'upsert', 'data': {'id': 8, 'name': 'C'}}],
+            [{'action': 'upsert', 'data': {'id': 7, 'name': 'B'}},
+             {'action': 'upsert', 'data': {'id': 9, 'name': 'D'}},
+             {'action': 'upsert', 'data': {'id': 10, 'name': 'E'}}]]
+
+        # Should be broken into 4 batches
+        self.assertEqual(len(target_stitch.OUR_SESSION.messages_sent), 4)
+
+        # Sort by length and remove sequence number to compare directly
+        actual_messages = [[{key: m[key] for key in ["action","data"]} for m in ms]
+                           for ms in sorted(target_stitch.OUR_SESSION.messages_sent, key=lambda ms: ms[0]['data']['id'])]
+
+        self.assertEqual(actual_messages, expected_messages)
+
+
+class BufferingPerStreamState(unittest.TestCase):
```

Comment:
I think this should mention that both tests fail their posts. Like `BufferingPerStreamNoStateOnFailure`

Reply:
Done.